core/src/main/java/org/infinispan/util/concurrent/BlockingManagerImpl.java

```diff
@@ -370,6 +370,19 @@ public <V> ScheduledBlockingCompletableStage<V> scheduleRunBlocking(Supplier<V>
       return scheduledStage;
    }
 
+   @Override
+   public ScheduledBlockingCompletableStage<Void> scheduleRunBlockingAtFixedRate(Runnable runnable, long initialDelay, long period, TimeUnit unit, Object traceId) {
+      Supplier supplier = () -> {
+         runnable.run();
+         return null;
+      };
+
+      var scheduledStage = new ScheduledBlockingFuture<>(supplier, traceId);
```

Comment:
This won't quite work as the stage will be completed on the first pass. We will need a slightly different abstraction since the task can be scheduled multiple times.

Reply:
I would say there is no reason for this to return a ScheduledBlockingCompletableStage as we don't ever expect to get a return value from it, unlike `scheduleRunBlocking`. In that case we can probably just return a `ScheduledFuture` like the `ScheduledExecutorService` does where get just always throws an exception but you can cancel it as needed. Then internally we can handle rescheduling the task as needed.

Reply:
you're definitively right.... fixing...